autoload/clap/provider/help_tags.vim

```diff
@@ -0,0 +1,67 @@
+" Author: Mark Wu <markplace@gmail.com>
+" Description: List the help tags.
+
+let s:save_cpo = &cpoptions
+set cpoptions&vim
+
+" Help tags support, ported from https://github.com/zeero/vim-ctrlp-help
+let s:help_tags_filecache = get(g:, 'clap_provider_help_tags_cache', fnamemodify($MYVIMRC, ':p:h') . '/clap_provider_help_tags_cache')
+let s:help_tags_memcache = []
+
+function! s:help_tags_source() abort
+  if empty(s:help_tags_memcache)
+    if getftime(s:help_tags_filecache) > max(map(s:get_tags_files(), 'getftime(v:val)'))
+      let s:help_tags_memcache = readfile(s:help_tags_filecache)
+    else
+      let s:help_tags_memcache = s:get_tags_list()
+      silent! call writefile(s:help_tags_memcache, s:help_tags_filecache)
+    endif
+  endif
+
+  return s:help_tags_memcache
+endfunction
+
+function! s:get_tags_list() abort
+  let tagsfiles = s:get_tags_files()
+
+  let input_dict = {}
+  for tagsfile in tagsfiles
+    for line in readfile(tagsfile)
+      let tag = split(line, "\t")
```

Comment:
`tags` are formated by tab, so this should be enough, no need to `matchstr()`.
```vim
      let items = split(line, "\t")
      let tag_subject = items[0]
      if !has_key(input_dict, tag_subject)
        let input_dict[tag_subject] = printf("%-60s\t%s", tag_subject, items[1])
      endif
```

Reply:
It's okay for me, I just modified it.